### PR TITLE
🏗️ Refactor/migrate base64 predicates

### PR DESCRIPTION
**Pull request by ccamel**
This PR continues the migration of predicates toward pure Prolog by portingbase64_encoded/3` and `base64url/2`  out of Go and into the new `base64.pl` Prolog library. `base64/2` has been removed (deprecated).
The implementation is fully in Prolog and does not require any VFS device.

Doc has been updated accordingly.

## Summary by CodeRabbit

* **Documentation**
  * Reorganized predicate library documentation with a new dedicated Base64 library category and reordered related library categories for improved navigation

* **Tests**
  * Expanded test coverage for Base64 encoding and decoding operations with new scenarios covering encoding options, decoding edge cases, and error handling

### CODE REVIEW

File: x/logic/lib/base64.pl [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+
+base64_option_match(Name, Option, Value) :-
+  nonvar(Option),
+  compound(Option),
+  Option =.. [Name, Value].
+
+base64_option_known(Option) :-
+  nonvar(Option),
+  compound(Option),
+  Option =.. [_Name, _Value].
```
[the remark this answers is not in the document]
Comment (ccamel) [reply x/logic/lib/base64.pl]: ah, good catch. What about a more idiomatic approach?

```prolog
base64_option_known(charset(_)).
base64_option_known(padding(_)).
base64_option_known(as(_)).
base64_option_known(encoding(_)).
```

File: x/logic/lib/base64.pl [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+  [C0, C1],
+  !,
+  {
+    base64_char(Charset, B0, C0),
+    base64_char(Charset, B1, C1),
+    A is (B0 << 18) + (B1 << 12),
+    I0 is (A >> 16) /\ 255
+  }.
+base64_decode_bytes(_, [], _) -->
+  [].
```
[the remark this answers is not in the document]
Comment (ccamel) [reply x/logic/lib/base64.pl]: hum, good catch. Will add validation to reject non-zero tail bits and stop decoding after padding. Will also add tests.

**Comment by ccamel** [comment]
@coderabbitai resolve
